be/src/runtime/exec_env.cpp

```diff
@@ -93,10 +92,8 @@ static int64_t calc_max_compaction_memory(int64_t process_mem_limit) {
     if (limit < 0) {
         limit = process_mem_limit;
     }
-    if (percent < 0 || percent > 100) {
-        percent = 100;
-    }
-    return std::min<int64_t>(limit, process_mem_limit * percent / 100);
+
+    return std::min<int64_t>(limit, process_mem_limit);
```

Comment:
current the config::compaction_memory_limit_per_worker is only used for change the chunk size, unable to limit the memory usage of one task.